pyaedt/modeler/cad/Primitives.py

```diff
@@ -5958,8 +5987,78 @@ def _create_region(self, pad_percent=300, is_percentage=True):
             "IsLightweight:=",
             False,
         ]
-        self.oeditor.CreateRegion(arg, arg2)
-        return self._create_object("Region")
+        return arg, arg2
+
+    @pyaedt_function_handler()
+    def _create_region(
+        self, pad_value=300, pad_type="Percentage Offset", region_name="Region", parts=None, region_type="Region"
+    ):
+        if region_name in self._app.modeler.objects_by_name:
+            self._app.logger.error("{} object already exists".format(region_name))
+            return False
+        if not isinstance(pad_value, list):
+            pad_value = [pad_value] * 6
+        is_percentage = pad_type in ["Percentage Offset", "Transverse Percentage Offset"]
+        arg, arg2 = self._parse_region_args(pad_value, pad_type, region_name, parts, region_type, is_percentage)
+        if arg and arg2:
+            self.oeditor.CreateRegion(arg, arg2)
+            return self._create_object(region_name)
+        else:
+            return False
+
+    @pyaedt_function_handler()
+    def create_region(self, pad_value=300, pad_type="Percentage Offset", region_name="Region", **kwarg):
+        """Create an air region.
+
+        Parameters
+        ----------
+        pad_value : float, str, list of floats or list of str, optional
+            Padding values to apply. If a list is not provided, the same
+            value is applied to all padding directions. If a list of floats
+            or strings is provided, the values are
+            interpreted as padding for ``["+X", "-X", "+Y", "-Y", "+Z", "-Z"]``.
+        pad_type : str, optional
+            Padding definition. The default is ``"Percentage Offset"``.
+            Options are ``"Absolute Offset"``,
+            ``"Absolute Position"``, ``"Percentage Offset"``, and
+            ``"Transverse Percentage Offset"``. When using a list,
+            different padding types can be provided for different
+           directions.
+        region_name : str, optional
+            Region name. The default is ``None``, in which case the name
+            is generated automatically.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.cad.object3d.Object3d`
+            Region object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateRegion
+        """
+        # backward compatibility
+        if kwarg:
+            if kwarg.get("is_percentage", False):
+                is_percentage = kwarg["is_percentage"]
+            else:
+                is_percentage = True
```

Comment:
```suggestion
            is_percentage =kwarg.get("is_percentage", False)
```
Not sure about this one as the code seems a bit confusing here. Do you mean that if you pass option `is_percentage=False` to the method, the variable should be equal to `True` inside ? I feel like the code will setup is_percentage to True whatever the value provided.

Reply:
You're absolutely right. The if statement is wrong.
The logic was supposed to reflect that `is_percentage` was set to `True` by default. So if it is not in the arguments, it should be set to True.
```suggestion
            if "is_percentage" in kwarg.keys():
                is_percentage = kwarg["is_percentage"]
            else:
                is_percentage = True
```